CMakeLists.txt

```diff
@@ -46,28 +46,39 @@ install(DIRECTORY launch config
   DESTINATION share/${PROJECT_NAME})
 
 if(BUILD_TESTING)
-  # find_package(roslint REQUIRED)
-  # roslint_cpp()
-  # roslint_python()
-  # roslint_add_test()
-
-  # find_package(roslaunch REQUIRED)
-  # roslaunch_add_file_check(launch/teleop.launch)
-
-  # find_package(rostest REQUIRED)
-  # include_directories(include ${catkin_INCLUDE_DIRS})
-
-  # # Check axes and scaling.
-  # add_rostest(test/differential_joy.test)
-  # add_rostest(test/holonomic_joy.test)
-  # add_rostest(test/six_dof_joy.test)
-
-  # # Check enable and turbo button logic.
-  # add_rostest(test/no_enable_joy.test)
-  # add_rostest(test/turbo_enable_joy.test)
-  # add_rostest(test/only_turbo_joy.test)
-  # add_rostest(test/turbo_angular_enable_joy.test)
-  # add_rostest(test/turbo_angular_enable_joy_with_rosparam_map.test)
+  # Need to specify LANGUAGE because hpp extension isn't used
```

Comment:
This is worth calling out - if the header used the `.hpp` extension instead of `.h`, we wouldn't need to call `ament_cppcheck` explicitly.

Reply:
Nevermind, this trick doesn't work well enough for me to suppress the implicit invocation but still call cppcheck explicitly. Still, cppcheck could be run if the header were renamed.
I'm going to call it out-of-scope for this PR though.

Reply:
Yeah, that's fine.  I'm going to to do a follow-up PR that does a bit more cleanup and enables composition.